### PR TITLE
Add support for remote controllers and delete old code

### PR DESCRIPTION
**Pull request by chuga**
- Delete "cloud mode" convenience initializers
- Delete FirebaseStorageController
- Delete RMXVariableFactory & |variableFromDictionary|
- Add RMXFirebaseRemoteController

### CODE REVIEW

File: src/core/RMXRemixer.m [hunk 1]
```diff
@@ -141,22 +130,6 @@ - (void)didSwipe:(UISwipeGestureRecognizer *)recognizer {
   [_overlayController showPanelAnimated:YES];
 }
 
-#pragma mark - Email
-
-+ (void)sendEmailInvite {
```
Comment (chriscox) [inline src/core/RMXRemixer.m:146]: I know this was hard-coded here... but will we still be able to provide some mechanism to get the generated device ID and firebase website to go to?

**Review by chriscox — CHANGES_REQUESTED** [review]
Since this is a breaking change, can you tag previous version to allow deprecation

File: src/core/RMXRemixer.m (hunk 1 above)
Comment (chuga) [reply src/core/RMXRemixer.m:146]: not as part of this PR (this was dead code)

**Review by chriscox — APPROVED** [review]
LGTM.

**Pull request merged**